app/src/pages/authCallbackPage/AuthCallbackPage.tsx

```diff
@@ -66,10 +61,30 @@ const AuthCallbackPage = (props: Props) => {
         const urlSearchParams = new URLSearchParams(window.location.search);
         const code = urlSearchParams.get('code') ?? '';
         const state = urlSearchParams.get('state') ?? '';
-        handleCallback({ baseUrl, code, state });
+        void handleCallback({ baseUrl, code, state });
     }, [baseUrl, setSession, navigate]);
 
-    return <Spinner status="Logging in..." />;
+    return shouldShowNonBSOLPage ? <NonBsolLandingPage /> : <Spinner status="Logging in..." />;
+};
+
+const userIsGpAdminNonBSOL = (auth: UserAuth | null): boolean => {
+    if (!auth) {
+        return false;
+    }
+
+    return auth.role === REPOSITORY_ROLE.GP_ADMIN && !auth.isBSOL;
+};
+
+const searchPatientPageByUserRole = (auth: UserAuth): routes => {
+    switch (auth?.role) {
+        case REPOSITORY_ROLE.GP_ADMIN:
```

Comment:
Should there be a case for this?

Reply:
yes, for both GP_ADMIN and GP_CLINICAL this would return routes.UPLOAD_SEARCH